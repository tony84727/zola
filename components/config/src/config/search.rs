use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
#[derive(Default)]
pub enum IndexFormat {
    ElasticlunrJson,
    #[default]
    ElasticlunrJavascript,
}

#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
#[serde(default)]
pub struct Search {
    /// Include the title of the page in the search index. `true` by default.
    pub include_title: bool,
    /// Includes the whole content in the search index. Ok for small sites but becomes
    /// too big on large sites. `true` by default.
    pub include_content: bool,
    /// Optionally truncate the content down to `n` chars. This might cut content in a word
    pub truncate_content_length: Option<usize>,
    /// Includes the description in the search index. When the site becomes too large, you can switch
    /// to that instead. `false` by default
    pub include_description: bool,
    /// Include the path of the page in the search index. `false` by default.
    pub include_path: bool,
    /// Foramt of the search index to be produced. Javascript by default
    pub index_format: IndexFormat,
}

impl Default for Search {
    fn default() -> Self {
        Search {
            include_title: true,
            include_content: true,
            include_description: false,
            include_path: false,
            truncate_content_length: None,
            index_format: Default::default(),
        }
    }
}

impl Search {
    pub fn serialize(&self) -> SerializedSearch {
        SerializedSearch { index_format: &self.index_format }
    }
}

#[derive(Serialize)]
pub struct SerializedSearch<'a> {
    pub index_format: &'a IndexFormat,
}
