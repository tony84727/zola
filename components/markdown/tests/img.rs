mod common;
use config::Config;

#[test]
fn can_transform_image() {
    let cases = vec![
        "![haha](https://example.com/abc.jpg)",
        "![](https://example.com/abc.jpg)",
        "![ha\"h>a](https://example.com/abc.jpg)",
        "![__ha__*ha*](https://example.com/abc.jpg)",
        "![ha[ha](https://example.com)](https://example.com/abc.jpg)",
    ];

    let body = common::render(&cases.join("\n")).unwrap().body;
    insta::assert_snapshot!(body);
}

#[test]
fn can_add_lazy_loading_and_async_decoding() {
    let cases = vec![
        "![haha](https://example.com/abc.jpg)",
        "![](https://example.com/abc.jpg)",
        "![ha\"h>a](https://example.com/abc.jpg)",
        "![__ha__*ha*](https://example.com/abc.jpg)",
        "![ha[ha](https://example.com)](https://example.com/abc.jpg)",
    ];

    let mut config = Config::default_for_test();
    config.markdown.lazy_async_image = true;

    let body = common::render_with_config(&cases.join("\n"), config).unwrap().body;
    insta::assert_snapshot!(body);
}
