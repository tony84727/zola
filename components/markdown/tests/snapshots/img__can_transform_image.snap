---
source: components/markdown/tests/img.rs
expression: body
---
<p><img src="https://example.com/abc.jpg" alt="haha" />
<img src="https://example.com/abc.jpg" alt="" />
<img src="https://example.com/abc.jpg" alt="ha&quot;h&gt;a" />
<img src="https://example.com/abc.jpg" alt="haha" />
<img src="https://example.com/abc.jpg" alt="haha" /></p>

